=== modified file 'IPython/deathrow/ipy_profile_zope.py'
--- IPython/Extensions/ipy_profile_zope.py	2009-09-18 02:40:20 +0000
+++ IPython/Extensions/ipy_profile_zope.py	2009-10-12 13:52:48 +0000
@@ -293,11 +293,6 @@
 
     zope_debug = ZopeDebug()
 
-    # <HACK ALERT>
-    import pdb;
-    pdb.set_trace = ipy_set_trace
-    # </HACK ALERT>
-
     # I like my banner minimal.
     o.banner = "ZOPE Py %s IPy %s\n" % (sys.version.split('\n')[0],release.version)
 

